omod-2.0/src/main/java/org/openmrs/module/attachments/rest/AttachmentResource2_0.java

```diff
@@ -1,17 +1,199 @@
+		        .property("birthdate", new DateProperty())
+		        .property("birthdateEstimated", new BooleanProperty()._default(false))
+		        .property("dead", new BooleanProperty()._default(false)).property("deathDate", new DateProperty())
+		        .property("causeOfDeath", new StringProperty())
+		        .property("addresses", new ArrayProperty(new RefProperty("#/definitions/AttachmentAddressCreate")))
+		        .property("attributes", new ArrayProperty(new RefProperty("#/definitions/AttachmentAttributeCreate")));
+		
+		model.setRequired(Arrays.asList("names", "gender"));
+		return model;
+	}
+	
+	@Override
+	public Model getUPDATEModel(Representation representation) {
+		return new ModelImpl().property("dead", new BooleanProperty()).property("causeOfDeath", new StringProperty())
+		        .property("deathDate", new DateProperty()).property("age", new IntegerProperty())
+		        .property("gender", new StringProperty()._enum("M")._enum("F")).property("birthdate", new DateProperty())
+		        .property("birthdateEstimated", new BooleanProperty()._default(false))
+		        .property("preferredName", new StringProperty().example("uuid"))
+		        .property("preferredAddress", new StringProperty().example("uuid"))
+		        .property("attributes", new ArrayProperty(new RefProperty("#/definitions/AttachmentAttributeCreate")))
+		        .required("dead").required("causeOfDeath");
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getPropertiesToExposeAsSubResources()
+	 */
+	@Override
+	public List<String> getPropertiesToExposeAsSubResources() {
+		return Arrays.asList("comment", "complexData", "Attributes");
+	}
+	
+	@Override
+	public Attachment newDelegate() {
+		return new Attachment();
+	}
+	
+	@SuppressWarnings("deprecation")
+	@Override
+	public Attachment getByUniqueId(String uniqueId) {
+		Obs obs = Context.getObsService().getObsByUuid(uniqueId);
+		if (!obs.isComplex()) {
+			throw new GenericRestException(uniqueId + " does not identify a complex obs.", null);
+		} else {
+			obs = Context.getObsService().getComplexObs(obs.getId(), AttachmentsConstants.ATT_VIEW_CRUD);
+			return new Attachment(obs);
+		}
+	}
+	
+	@Override
+	protected void delete(Attachment delegate, String reason, RequestContext context) throws ResponseException {
+		String encounterUuid = delegate.getObs().getEncounter() != null ? delegate.getObs().getEncounter().getUuid() : null;
+		Context.getObsService().voidObs(delegate.getObs(), REASON);
+		voidEncounterIfEmpty(Context.getEncounterService(), encounterUuid);
+	}
+	
+	@Override
+	public void purge(Attachment delegate, RequestContext context) throws ResponseException {
+		String encounterUuid = delegate.getObs().getEncounter() != null ? delegate.getObs().getEncounter().getUuid() : null;
+		Context.getObsService().purgeObs(delegate.getObs());
+		voidEncounterIfEmpty(Context.getEncounterService(), encounterUuid);
+	}
```

Comment:
Is this also required for Swagger documentation? If not, why did you override this? I see your doing great stuff here but I think it maybe out of scope. 

Same case for: `newDelegate(...)`, `delete(...)`, `getByUniqueId(...)`

Reply:
@samuelmale  thanks , let me cross check that code

Reply:
I have resolved that , take a look